EntAppProject/src/main/java/com/jobook/EntAppProject/Services/JobServiceStub.java

```diff
@@ -9,49 +9,56 @@
 import com.jobook.EntAppProject.model.Customer;
 import com.jobook.EntAppProject.model.Employee;
 import com.jobook.EntAppProject.model.Job;
+import org.springframework.util.Assert;
 
 @Component
 public class JobServiceStub implements IJobService {
-	
+
+	private final IUserService userService;
+
+	private List<Job> jobs;
+
 	@Autowired
-	IUserService userService;
-	
-	private List<Job> jobs; 
-	
+	public JobServiceStub(IUserService userService){
+		this.userService = userService;
+	}
+
+
 	private void generateTestJobs()
 	{
-		jobs = new ArrayList<Job>();
+		Assert.notNull(userService, "User service is null -- check DI container");

```

Comment:
When using dependency injection, ALWAYS check to make sure that your dependency is actually injected/instantiated. Otherwise you the run risk of someone blindly creating new JobServiceStub(null) or something like that without injection, and it creates exceptions. This makes it more clear to the developer where they messed up.

Reply:
Good callout here.